tlv/primitive.go

```diff
@@ -195,6 +222,18 @@ func DUint64(r io.Reader, val interface{}, buf *[8]byte, l uint64) error {
 	return NewTypeForDecodingErr(val, "uint64", l, 8)
 }
 
+// DBool decodes a boolean. An error is returned if val is not a boolean.
+func DBool(r io.Reader, val interface{}, buf *[8]byte, l uint64) error {
+	if i, ok := val.(*bool); ok && l == 1 {
+		if _, err := io.ReadFull(r, buf[:1]); err != nil {
+			return err
+		}
+		*i = buf[0] != 0
```

Comment:
Maybe we should return an error if `val` is not 0 or 1... If that happens when decoding from the DB it's a strong indication that there's a bug or data corruption somewhere.

Reply:
Thanks for the review, pushed new changes with respect to this.